src/shell/shell.c

```diff
@@ -277,6 +285,9 @@ int main (int argc, char *argv[])
     if (!(shell.io = shell_io_create (shell.h, shell.info)))
         log_err_exit ("shell_io_create");
 
+    if (shell_pre_exec (&shell) < 0)
```

Comment:
Were you wanting to move all pre-exec initialization to this function?  (Fine by me)

Reply:
I wasn't thinking of moving all pre-exec initialization to this function, but was thinking the `shell_pre_exec()` function as a start of a hook for plugins. I was thinking that the signals and kill-event handlers setup fit well into this model as kind of internal plugins. The other initialization being done in the shell somehow seemed more fundamental, so didn't lend itself to easily move into this function. However, maybe at least pmi initialization should go in there...

How about I tack this on to the end once I've had a chance to fully complete this feature (starting to get lost with the number of fixup commits...)

Reply:
Glad you're forward thinking about plugins!   Sure, however you like.